README.md

```diff
@@ -120,3 +120,152 @@ the components involved in the test:
 ``
 cd test/integration && dlv test -- -test.run ^TestCrud$
 ``
+
+#### E2E
+
+The fnord E2E tests can run in an unmanaged, managed, or hybrid modes. For both
+unmanaged and hybrid modes, you will need to bring your own clusters (BYOC).
+The managed mode runs similarly to integration as it uses the same test fixture
+setup.
+
+##### Managed
+
+The E2E managed tests will spin up the same environment as the
+[Integration](README.md#integration) tests described above and run [CRUD
+(create-read-update-delete)
+checks](https://github.com/marun/fnord/blob/master/test/e2e/crud.go) for
+federated types against that federation. To run:
+
+ - ensure the same binaries are available as described in the
+   [Integration](README.md#integration) section.
+ - `cd test/e2e && go test -i && go test -v`
+
+To run tests for a single type:
+
+```
+cd test/e2e && go test -i && go test -v --ginkgo.focus='"FederatedSecret"'
+```
+
+It may be helpful to use the [delve
+debugger](https://github.com/derekparker/delve) to gain insight into
+the components involved in the test:
+
+```
+cd test/e2e && dlv test -- -v=4 -test.v --ginkgo.focus='"FederatedSecret"'
+```
+
+##### Unmanaged and Hybrid Cluster Setup
+
+###### Create Clusters
+
+The quickest way to set up clusters for use with unmanaged and hybrid modes is
+to use [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/).
+Once you have minikube installed run:
+
+```
+minikube start -p clusterA --vm-driver=kvm
+```
+
+###### Deploy the Cluster Registry
+
+Get the latest version of the [Cluster
+Registry](https://github.com/kubernetes/cluster-registry/releases) and run the
+following:
+
+```
+crinit aggregated init mycr --host-cluster-context=clusterA
+```
+
+You can also specify your own image using the `--image` flag.
+
+###### Deploy Federation
+
+First you'll need to create the namespace to deploy the federation into:
+
+```
+kubectl create ns federation
+```
+
+Then run `apiserver-boot` to deploy:
+
+```
+apiserver-boot run in-cluster --name federation --namespace federation \
+    --image <containerregistry>/<username>/<imagename>:<tagname> \
+    --controller-args="-logtostderr,-v=4"
+```
+
+From here, the Unmanaged and Hybrid setups differ slightly. Proceed to the
+corresponding subsection depending on what you're interested in. If you are unsure,
+the Hybrid setup is best for debugging as you can run the controllers
+in-process with delve.
+
+###### Unmanaged
+
+1. Build kubefnord
+    ```
+    go build -o bin/kubefnord cmd/kubefnord/kubefnord.go
+
+    ```
+1. Join Cluster(s)
+    ```
+    ./bin/kubefnord join clusterA --host-cluster-context clusterA --add-to-registry --v=2
+    ```
+1. Run tests
+    ```
+    cd test/e2e
+    go test -c -v
+    ./e2e.test -v=4 -kubeconfig=/path/to/kubeconfig -test.v --ginkgo.focus='"FederatedSecret" resources'
+    ```
+
+You can repeat these steps to join any additional clusters.
+
+###### Hybrid
+
```

Comment:
Can you explain what is meant by "hybrid" here?

Reply:
Yup! This was still WIP so did not elaborate and instead just provided the commands so others could get started.